src/metabase/analyze/fingerprint/insights.clj

```diff
@@ -225,15 +224,13 @@
   (m/find-first (partial valid-period? from to) (keys unit->duration)))
 
 (defn- ->millis-from-epoch [t]
-  (when t
-    (condp instance? t
-      Instant        (.toEpochMilli ^Instant t)
-      OffsetDateTime (.toEpochMilli (.toInstant ^OffsetDateTime t))
-      ZonedDateTime  (.toEpochMilli (.toInstant ^ZonedDateTime t))
-      LocalDate      (->millis-from-epoch (t/offset-date-time t (t/local-time 0) (t/zone-offset 0)))
-      LocalDateTime  (->millis-from-epoch (t/offset-date-time t (t/zone-offset 0)))
-      LocalTime      (->millis-from-epoch (t/offset-date-time (t/local-date "1970-01-01") t (t/zone-offset 0)))
-      OffsetTime     (->millis-from-epoch (t/offset-date-time (t/local-date "1970-01-01") t (t/zone-offset t))))))
+  (cond (instance? Instant t)        (.toEpochMilli ^Instant t)
```

Comment:
Now that we're using a `cond` it might be a good idea to add a catch-all case for any other non-nil type, throwing an error rather than returning nil.